projects/batfish/src/main/java/org/batfish/bdp/VirtualRouter.java

```diff
@@ -217,11 +219,50 @@ public void computeFib() {
     _fib = new Fib(_mainRib);
   }
 
-  public boolean computeInterAreaSummaries() {
+  /**
+   * Decides whether the current OSPF summary route metric needs to be changed based on the given
+   * route's metric.
+   *
+   * <p>Routes from the same area or outside of areaPrefix have no effect on the summary metric.
+   *
+   * @param route The route in question, whose metric is considered
+   * @param areaPrefix The Ip prefix of the OSPF area
+   * @param currentMetric The current summary metric for the area
+   * @param areaNum Area number.
+   * @return the newly computed summary metric.
+   */
+  @Nullable
+  static Long computeUpdatedOspfSummaryMetric(
+      OspfAreaRoute route, Prefix areaPrefix, @Nullable Long currentMetric, long areaNum) {
+    Prefix contributingRoutePrefix = route.getNetwork();
```

Comment:
I generally prefer to structure my code with "exit early". e.g.,

```java
if (areaNum == route.getArea()) {
    // Routes from the same area have no effect.
    return currentMetric;
}

Prefix contributingRoutePrefix = route.getNetwork();
if (!areaPrefix.containsPrefix(contributingRoutePrefix)) {
    // Routes from outside of teh area prefix have no effect.
    return currentMetric;
}

...
```

This also makes the code generally less nested, hence easier to read.

Reply:
(obviously you decide whether these are different `if` or the same `if`, etc.)